### PR TITLE
private-org-sync: use authentication for sources

### PR DESCRIPTION
**Pull request by petr-muller**
This is necessary to support private sources.

/cc @droslean

### CODE REVIEW

**Comment by petr-muller** [comment]
This should fix the failure to sync openshift/openshift-private-tests:

https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/logs/periodic-openshift-release-private-org-sync/1273224025983684608#1:build-log.txt%3A419

File: cmd/private-org-sync/main.go [hunk 1]
```diff
@@ -421,6 +421,9 @@ func addGitRemote(logger *logrus.Entry, git gitFunc, org, repo, repoDir, remoteN
 		remoteSetupLogger.WithError(err).Error("Failed to construct URL for the source remote")
 		return fmt.Errorf("failed to construct URL for the source remote")
 	}
+	if token != "" {
+		srcURL.User = url.User(token)
```
Comment (droslean) [inline cmd/private-org-sync/main.go:425]: Just to make sure, do we censor this?

Comment (petr-muller) [reply cmd/private-org-sync/main.go:425]: We should, there is the censoring logger setup, and we are already using tokens in the destination remotes.

But I'll explicitly check this new usage, better to be sure about this.

/hold

**Comment by petr-muller** [comment]
Tested the censoring, looks good:

```console
$ go run ./cmd/private-org-sync/ --config-path ~/Temporary/empty-directory/ --whitelist-file whitelist.yaml --target-org muller-testing-org --git-name "Petr Muller" --git-email "muller@redhat.com" --token-path=token --debug
DEBU[0000] Running git                                   command="git ls-remote --heads https://CENSORED@github.com/openshift/openshift-tests-private" remote="https://CENSORED@github.com/openshift/openshift-tests-private"
DEBU[0003] Executed command                              command="git ls-remote --heads https://CENSORED@github.com/openshift/openshift-tests-private" output="a84cebcf0bdbd5cb932f638083474462555377a7\trefs/heads/master\nbb5994f545ad537206c2ccda14d8bac2dad07032\trefs/heads/release-4.4\nb38546362d581936476d1ebf4399058e7c644fa8\trefs/heads/release-4.5\na84cebcf0bdbd5cb932f638083474462555377a7\trefs/heads/release-4.6\na84cebcf0bdbd5cb932f638083474462555377a7\trefs/heads/release-4.7\n" remote="https://CENSORED@github.com/openshift/openshift-tests-private"
INFO[0003] Syncing content between locations             branch=release-4.7 destination=muller-testing-org/openshift-tests-private@release-4.7 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.7 source-file=openshift-openshift-tests-private-release-4.7.yaml
DEBU[0003] Determining HEAD of destination branch        branch=release-4.7 destination=muller-testing-org/openshift-tests-private@release-4.7 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.7 source-file=openshift-openshift-tests-private-release-4.7.yaml
DEBU[0003] Running git                                   branch=release-4.7 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.7 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.7 source-file=openshift-openshift-tests-private-release-4.7.yaml
DEBU[0003] Git command was executed but completed with non-zero exit code  branch=release-4.7 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.7 error="exit status 128" local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift output="remote: Repository not found.\nfatal: repository 'https://github.com/muller-testing-org/openshift-tests-private/' not found\n" repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.7 source-file=openshift-openshift-tests-private-release-4.7.yaml
WARN[0003] destination repository does not exist or we cannot access it  branch=release-4.7 destination=muller-testing-org/openshift-tests-private@release-4.7 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.7 source-file=openshift-openshift-tests-private-release-4.7.yaml
INFO[0003] Syncing content between locations             branch=master destination=muller-testing-org/openshift-tests-private@master local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@master source-file=openshift-openshift-tests-private-master.yaml
DEBU[0003] Determining HEAD of destination branch        branch=master destination=muller-testing-org/openshift-tests-private@master local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@master source-file=openshift-openshift-tests-private-master.yaml
DEBU[0003] Running git                                   branch=master command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@master local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@master source-file=openshift-openshift-tests-private-master.yaml
DEBU[0004] Git command was executed but completed with non-zero exit code  branch=master command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@master error="exit status 128" local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift output="remote: Repository not found.\nfatal: repository 'https://github.com/muller-testing-org/openshift-tests-private/' not found\n" repo=openshift-tests-private source=openshift/openshift-tests-private@master source-file=openshift-openshift-tests-private-master.yaml
WARN[0004] destination repository does not exist or we cannot access it  branch=master destination=muller-testing-org/openshift-tests-private@master local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@master source-file=openshift-openshift-tests-private-master.yaml
INFO[0004] Syncing content between locations             branch=release-4.4 destination=muller-testing-org/openshift-tests-private@release-4.4 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.4 source-file=openshift-openshift-tests-private-release-4.4.yaml
DEBU[0004] Determining HEAD of destination branch        branch=release-4.4 destination=muller-testing-org/openshift-tests-private@release-4.4 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.4 source-file=openshift-openshift-tests-private-release-4.4.yaml
DEBU[0004] Running git                                   branch=release-4.4 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.4 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.4 source-file=openshift-openshift-tests-private-release-4.4.yaml
DEBU[0005] Git command was executed but completed with non-zero exit code  branch=release-4.4 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.4 error="exit status 128" local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift output="remote: Repository not found.\nfatal: repository 'https://github.com/muller-testing-org/openshift-tests-private/' not found\n" repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.4 source-file=openshift-openshift-tests-private-release-4.4.yaml
WARN[0005] destination repository does not exist or we cannot access it  branch=release-4.4 destination=muller-testing-org/openshift-tests-private@release-4.4 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.4 source-file=openshift-openshift-tests-private-release-4.4.yaml
INFO[0005] Syncing content between locations             branch=release-4.5 destination=muller-testing-org/openshift-tests-private@release-4.5 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.5 source-file=openshift-openshift-tests-private-release-4.5.yaml
DEBU[0005] Determining HEAD of destination branch        branch=release-4.5 destination=muller-testing-org/openshift-tests-private@release-4.5 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.5 source-file=openshift-openshift-tests-private-release-4.5.yaml
DEBU[0005] Running git                                   branch=release-4.5 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.5 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.5 source-file=openshift-openshift-tests-private-release-4.5.yaml
DEBU[0007] Git command was executed but completed with non-zero exit code  branch=release-4.5 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.5 error="exit status 128" local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift output="remote: Repository not found.\nfatal: repository 'https://github.com/muller-testing-org/openshift-tests-private/' not found\n" repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.5 source-file=openshift-openshift-tests-private-release-4.5.yaml
WARN[0007] destination repository does not exist or we cannot access it  branch=release-4.5 destination=muller-testing-org/openshift-tests-private@release-4.5 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.5 source-file=openshift-openshift-tests-private-release-4.5.yaml
INFO[0007] Syncing content between locations             branch=release-4.6 destination=muller-testing-org/openshift-tests-private@release-4.6 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.6 source-file=openshift-openshift-tests-private-release-4.6.yaml
DEBU[0007] Determining HEAD of destination branch        branch=release-4.6 destination=muller-testing-org/openshift-tests-private@release-4.6 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.6 source-file=openshift-openshift-tests-private-release-4.6.yaml
DEBU[0007] Running git                                   branch=release-4.6 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.6 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.6 source-file=openshift-openshift-tests-private-release-4.6.yaml
DEBU[0008] Git command was executed but completed with non-zero exit code  branch=release-4.6 command="git ls-remote --heads https://CENSORED@github.com/muller-testing-org/openshift-tests-private" destination=muller-testing-org/openshift-tests-private@release-4.6 error="exit status 128" local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift output="remote: Repository not found.\nfatal: repository 'https://github.com/muller-testing-org/openshift-tests-private/' not found\n" repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.6 source-file=openshift-openshift-tests-private-release-4.6.yaml
WARN[0008] destination repository does not exist or we cannot access it  branch=release-4.6 destination=muller-testing-org/openshift-tests-private@release-4.6 local-repo=/tmp/374777101/openshift/openshift-tests-private org=openshift repo=openshift-tests-private source=openshift/openshift-tests-private@release-4.6 source-file=openshift-openshift-tests-private-release-4.6.yaml
```

**Comment by petr-muller** [comment]
I needed to add more code to handle the initial branch detection for the allowlisted repos

**Pull request merged**